src/syntax/FStar.Syntax.Resugar.fs

```diff
@@ -817,6 +839,130 @@ and resugar_match_pat (p:S.pat) : A.pattern =
     | Pat_dot_term (bv, term) ->
       // no sure if this is correct resugar since the term is not generated from desugar
       let pat = mk (A.PatVar(bv_as_unique_ident bv, None)) in
-      mk (A.PatAscribed(pat, resugar_term term))
+      if Options.print_bound_var_types() then
+        mk (A.PatAscribed(pat, resugar_term term))
+      else
+        pat
   in
     aux p
+
+let resugar_qualifier : S.qualifier -> option<A.qualifier> = function
+  | S.Assumption -> Some A.Assumption
+  | S.New -> Some A.New
+  | S.Private -> Some A.Private
+  | S.Unfold_for_unification_and_vcgen -> Some A.Unfold_for_unification_and_vcgen
+  (* TODO : Find the correct option to display this *)
+  | Visible_default -> if true then None else Some A.Visible
+  | S.Irreducible -> Some A.Irreducible
+  | S.Abstract -> Some A.Abstract
+  | S.Inline_for_extraction -> Some A.Inline_for_extraction
+  | S.NoExtract -> Some A.NoExtract
+  | S.Noeq -> Some A.Noeq
+  | S.Unopteq -> Some A.Unopteq
+  | S.TotalEffect -> Some A.TotalEffect
+  (* TODO : Find the correct option to display this *)
+  | S.Logic -> if true then None else Some A.Logic
+  | S.Reifiable -> Some A.Reifiable
+  | S.Reflectable _ -> Some A.Reflectable
+  | S.Discriminator _ -> None
+  | S.Projector _ -> None
+  | S.RecordType _ -> None
+  | S.RecordConstructor _ -> None
+  | S.Action _ -> None
+  | S.ExceptionConstructor -> None
+  | S.HasMaskedEffect -> None
+  | S.Effect -> Some A.Effect_qual
+  | S.OnlyName -> None
+
+
+let resugar_pragma = function
+  | S.SetOptions s -> A.SetOptions s
+  | S.ResetOptions s -> A.ResetOptions s
+  | S.LightOff -> A.LightOff
+
+let resugar_typ datacon_ses se : sigelts * A.tycon =
+  match se.sigel with
+  | Sig_inductive_typ (tylid, uvs, bs, t, _, datacons) ->
+      let current_datacons, other_datacons = datacon_ses |> List.partition (fun se -> match se.sigel with
+        | Sig_datacon (_, _, _, inductive_lid, _, _) -> lid_equals inductive_lid tylid
+        | _ -> failwith "unexpected" )
+      in
+      assert (List.length current_datacons = List.length datacons) ;
+      let tyc =
+        if se.sigquals |> BU.for_some (function | RecordType _ -> true | _ -> false)
+        then
+          (* Resugar as a record *)
+          (failwith "NIY")
+        else
+          (* Resugar as a variant *)
+          (failwith "NIY")
+      in
+      other_datacons, tyc
+  | Sig_declare_typ (tylid, uvs, t) ->
+      (failwith "NIY")
+  | _ -> failwith "Impossible : only Sig_inductive_typ and Sig_declare_typ can be resugared as types"
+
+let decl'_to_decl se d' =
+  {
+    d = d' ;
+    drange = se.sigrng ;
+    (* TODO : documentation should be retrieved from the desugaring environment at some point *)
+    doc = None ;
+    quals = List.choose resugar_qualifier se.sigquals ;
+    (* TODO : are these stocked up somewhere ? *)
+    attrs = [] ;
+  }
+
+let resugar_sigelt se : option<A.decl> =
+  match se.sigel with
```

Comment:
There is no pressing issue. It is just that I said I would fix the inconsistency in print out the module name, but I think it can wait until the sigelts are complete.